packages/storage-plus/src/de.rs

```diff
@@ -0,0 +1,119 @@
+use crate::keys::{IntKey, TimestampKey};
+use cosmwasm_std::{Addr, StdError, StdResult};
+use std::array::TryFromSliceError;
+use std::convert::TryInto;
+
+pub trait KeyDeserialize {
+    type Output: Sized;
+
+    fn from_slice(value: &[u8]) -> StdResult<Self::Output>;
+}
+
+impl KeyDeserialize for () {
+    type Output = ();
+
+    fn from_slice(_value: &[u8]) -> StdResult<Self::Output> {
+        Ok(())
+    }
+}
+
+macro_rules! bytes_de {
+    (for $($t:ty),+) => {
+        $(impl KeyDeserialize for $t {
+            type Output = Vec<u8>;
+
+            fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+                Ok(value.to_vec())
+            }
+        })*
+    }
+}
+
+bytes_de!(for Vec<u8>, &Vec<u8>, [u8], &[u8]);
+
+macro_rules! string_de {
+    (for $($t:ty),+) => {
+        $(impl KeyDeserialize for $t {
+            type Output = String;
+
+            fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+                // FIXME?: Use `from_utf8_unchecked` for String, &str
```

Comment:
Why unchecked? And what if for some strange reasons underlying think is not exactly utf (malformed data, some kind of attack, whatever). Iteration through couple character versus cost of loading data from storage is irrelevant.

Reply:
OK. Let's stick to `from_utf8`. Will remove this comment.